### DISCUSSION TITLE
Valetudo 2022.08.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1537] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2022.08.0</h2></p>
</div>

New firmwares for dreame robots and some bugfixes

## New dreame firmwares

[<img src="https://user-images.githubusercontent.com/974410/182454333-356ba835-4f69-4041-a720-7c2d97eb3209.png" width=350>](https://user-images.githubusercontent.com/974410/182454333-356ba835-4f69-4041-a720-7c2d97eb3209.png)

It took quite a lot of time and work, but now we're proud to announce that we've managed to get newer dreame firmwares
to work rooted and with Valetudo. A big thank you to everyone who helped during the beta test!

**Important**: That does **not** mean that these newer firmwares have become rootable.
You will need to have an already rooted robot to install these latest rooted firmwares.

To update, head over to the dustbuilder at [https://builder.dontvacuum.me/](https://builder.dontvacuum.me/) to build and install a new image.
Make sure to update Valetudo **before** updating the firmware.

## Other dreame news

[<img src="https://user-images.githubusercontent.com/974410/182454516-57ef159d-d97e-458a-815d-97a60db689ba.png" width=350>](https://user-images.githubusercontent.com/974410/182454516-57ef159d-d97e-458a-815d-97a60db689ba.png)

Starting today, every firmware image built with the dustbuilder will contain a fix for the vanishing Valetudo issue,
which occasionally left people with a rooted robot but no Valetudo.

[<img src="https://user-images.githubusercontent.com/974410/182454599-cab62e88-6179-43da-b907-6ac7236c7d17.png" width=350>](https://user-images.githubusercontent.com/974410/182454599-cab62e88-6179-43da-b907-6ac7236c7d17.png)

If a user starts a new mapping pass, the fresh map should now appear instantly.
Furthermore, issues with frozen maps either during cleanups or when editing segments were fixed as well.

## Robot Coverage View

[<img src="https://user-images.githubusercontent.com/974410/182455081-879226c5-aa14-4160-bc2f-f9c0f2a2d3d3.png" width=300>](https://user-images.githubusercontent.com/974410/182455081-879226c5-aa14-4160-bc2f-f9c0f2a2d3d3.png)

The UI has been extended with a view displaying a much thicker path.
This can be used to better see which areas of the map have been cleaned.

Whether this view will stick around long-term is yet to be decided. It was pretty easy to implement though.
Feel free to leave your feedback in the comments.

## The usual

[<img src="https://user-images.githubusercontent.com/974410/182454845-cc693f37-a126-4978-b2f0-5c9f77a970d8.png" width=350>](https://user-images.githubusercontent.com/974410/182454845-cc693f37-a126-4978-b2f0-5c9f77a970d8.png)

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating if you haven't done so already:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

## Misc

This changelog features artworks generated by DALL-E 2 because I thought it would be neat to have something to look at.
Unfortunately, it doesn't like generating artworks featuring lidar-based vacuum robots (yet?)

I hope they've nonetheless enhanced your changelog reading experience.
Feel free to leave feedback on that as well.

### Features

- **vendor.dreame**: Newer D9 Firmwares also have a sensor consumable [`17c0d27`](http://github.com/Hypfer/Valetudo/commit/17c0d27fb69b92b5a8028d6bab489e5b97fb4537)
- **ui**: Trim whitespaces when renaming segments [`39d72f4`](http://github.com/Hypfer/Valetudo/commit/39d72f4e787108c75a1cb501995151fb19b4d8b1)
- **ui**: Add robot coverage map [`a9661a0`](http://github.com/Hypfer/Valetudo/commit/a9661a02bcbe642c3e7de46be4015f9532e4ce49)
- **vendor.dreame**: Add mop-only quirk for newer firmwares [`a33afe3`](http://github.com/Hypfer/Valetudo/commit/a33afe351c1b868919bc7b036e2e02a05b8e581c)
- **vendor.dreame**: Add Mop Dock UV-C Toggle Quirk [`7ce4abe`](http://github.com/Hypfer/Valetudo/commit/7ce4abe7353567fd3db6a7a9cc1c767d68a9db93)

### Fixes

- **vendor.dreame**: Add support for more features of newer firmwares [`b74c281`](http://github.com/Hypfer/Valetudo/commit/b74c281e1727ca2a007d96915c6fceb3f386bd28)
- **ui**: Add margin to apple touch icon (#1526) [`dc72ef0`](http://github.com/Hypfer/Valetudo/commit/dc72ef05319799020a504a3b20e0b7e6092984ee)
- **vendor.dreame**: Ignore empty maps [`99aad7b`](http://github.com/Hypfer/Valetudo/commit/99aad7be8b20c9459e849dd6b9ce8df5171b9690)
- **networkadvertisement**: Don't crash the process on ssdp multicast membership errors [`bfb96c0`](http://github.com/Hypfer/Valetudo/commit/bfb96c09aa2bc0ff9e151fb6cdab4ac5bbd58a36)
- **miio**: Allow two simultaneous FDS uploads [`6e26374`](http://github.com/Hypfer/Valetudo/commit/6e26374c32e9fe33c93f4f91c189f917c5f884a5)
- **vendor.dreame**: Switch to async map preprocessing to not block the event loop [`3b5743b`](http://github.com/Hypfer/Valetudo/commit/3b5743b41264c3d8d36d8cc0a75da30e90bb7338)
- **miio**: Further improve fds upload timeout handling [`8c1ba9e`](http://github.com/Hypfer/Valetudo/commit/8c1ba9ea2fcc980bf726a4678902afd2577a0191)
- **miio**: Gracefully handle connectivity issues when receiving fds uploads [`8aab94f`](http://github.com/Hypfer/Valetudo/commit/8aab94fa2ab01479997968904e9312fbc8111a83)
- **vendor.dreame**: Ignore timezone update message [`5ecf7fa`](http://github.com/Hypfer/Valetudo/commit/5ecf7fa0842c498a2ad8abfd792a71b420ce34bf)
- **ui**: Allow sending home basic control command while paused [`0dfb4ba`](http://github.com/Hypfer/Valetudo/commit/0dfb4ba59bd1334f994beea211ac17fdf5323f1f)
- **vendor.dreame**: Fix maps for newer firmwares [`5f67103`](http://github.com/Hypfer/Valetudo/commit/5f6710348fd304046a201c2423490555e44d3b20)
- **webserver**: Fix openapi schema for MapSegmentEditCapability [`f106735`](http://github.com/Hypfer/Valetudo/commit/f106735cbae0eed2ebfebaa6843c7da976c5127e)

### Refactoring

- **webserver**: Response cleanup [`839310a`](http://github.com/Hypfer/Valetudo/commit/839310ab0941941d68ea4ba0c1fbfc17b8924443)
- **webserver**: Actually make use of the openapi schema validation [`6e2d2e1`](http://github.com/Hypfer/Valetudo/commit/6e2d2e1904040aa891dda5d35909654553e04eb7)
- Introduce RobotFirmwareError and refactor CapabilityRouter for unified error logs [`1d2ad47`](http://github.com/Hypfer/Valetudo/commit/1d2ad47e9d49eac71f0621d2aa0444acf871acec)

### Chores

- **release**: 2022.08.0 [`214b5c0`](http://github.com/Hypfer/Valetudo/commit/214b5c0c11c0cf70fa56ff915fda05e064a18355)
- Minor cleanup [`9356ba8`](http://github.com/Hypfer/Valetudo/commit/9356ba8e19825a1602cd842db9487389aa553ccf)
- **vendor.dreame**: Fix tests [`15eee37`](http://github.com/Hypfer/Valetudo/commit/15eee3749e41fc77ddf62fbaf8fc348f8d872672)
- **mqtt**: Minor cleanup [`4ae1d03`](http://github.com/Hypfer/Valetudo/commit/4ae1d0365d2b54f9d27c49caf5e38b915ff47e4c)
- Minor misc cleanups [`7bf3c87`](http://github.com/Hypfer/Valetudo/commit/7bf3c87db34a8f27624fe735737a4ccd17fbf8bc)
- Add eslintrc flavors [`b3843f8`](http://github.com/Hypfer/Valetudo/commit/b3843f8bd3c95346735d674f7df4543ff5f78b81)
- Add close_threads github workflow [`e10313d`](http://github.com/Hypfer/Valetudo/commit/e10313dc75a0f7867d64c195fbd9ef69ca3b199b)
- **ui**: Add missing rel="noopener" tag to menu bar [`a7e6920`](http://github.com/Hypfer/Valetudo/commit/a7e6920c7924b5db43b9eb086a434fddaa276d75)
- **build**: Bump to NodeJS v18.5.0 [`66712f9`](http://github.com/Hypfer/Valetudo/commit/66712f91dd3f16c24b0b22b334a37ecfeb983df9)
- Update wording in VoicepackHelp.ts [`05a6b89`](http://github.com/Hypfer/Valetudo/commit/05a6b89848b5c1e9a04f4a822da58e03cd89b5f8)
- Update issue template [`cebca05`](http://github.com/Hypfer/Valetudo/commit/cebca057ceb05d202f3f76211a2e63290f39b002)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2022.08.0'>Valetudo 2022.08.0</a>.</em>

**Comment by thetic** [comment]
The coverage view looks great, but it does make it hard to discern the actual path the robot took. I think displaying the old, thin line path _on top of_ the new coverage display would give us the best of both worlds.

**Reply by Hypfer** [reply]
![image](https://user-images.githubusercontent.com/974410/183131997-db617ac6-b7be-4bd8-89f5-6b18f4309084.png)

![image](https://user-images.githubusercontent.com/974410/183132004-f1a85958-0096-46e8-9ebb-3117b52b30dc.png)


Like this?

**Reply by mrJudahBella** [reply]
It would be nice if the coverage path is semi-transparent as well, to present some kind of context as to what was cleaned (the underlying map). If not, it's hard to make out room details and boundaries, at least for me.

**Comment by rwjack** [comment]
When upgrading the Z10:
- I first went to it's web UI and updated valetudo to 2022.08.0
- I'm then supposed to make a build job with the new firmware on builder.dontvacuum.me
- If so, do I prepackage valetudo or is there no need since it is already installed?
![image](https://user-images.githubusercontent.com/59068073/183287624-3ca1843a-d297-4b18-9562-591170755583.png)

- also, the A/B filesystem part had me confused even when doing the initial root (I just did everything twice as described [here 🦆](https://valetudo.cloud/pages/general/rooting-instructions.html).
- Does the filesystem change itself on each reboot A->B and B->A or do I need to trigger it somehow?

**Reply by rwjack** [reply]
For now I simply upgraded valetudo via web ui, didn't prepackage valetudo on the builder, just downloaded the latest FW and did `install.sh` once.

Do I need to do anything else?

**Reply by Hypfer** [reply]
> Do I need to do anything else?

Nope that's all
